app/controllers/openid_connect/authorization_controller.rb

```diff
@@ -23,9 +23,13 @@ def index
 
     def create
       result = @authorize_form.submit(current_user, session.id)
-      analytics.track_event(Analytics::OPENID_CONNECT_ALLOW, result.except(:redirect_uri))
+      analytics_attributes = result.to_h
 
-      if (redirect_uri = result[:redirect_uri])
+      analytics.track_event(
+        Analytics::OPENID_CONNECT_ALLOW, analytics_attributes.except(:redirect_uri)
+      )
+
+      if (redirect_uri = analytics_attributes[:redirect_uri])
```

Comment:
I thought about making it a separate method, but I don't think that's the right thing to do because that means the `redirect_uri` can be accessed outside the scope of submission, which shouldn't be allowed. I think the best thing is to modify FormResponse such that `extra` is a public method.